edb/schema/constraints.py

```diff
@@ -372,27 +468,43 @@ def _constraint_args_from_ast(cls, schema, astnode, context):
 
         return args
 
-    def compile_expr_field(self, schema, context, field, value):
+    def compile_expr_field(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+        field: so.Field[Any],
+        value: s_expr.Expression,
+    ) -> s_expr.Expression:
         from edb.edgeql import compiler as qlcompiler
 
         if field.name in ('expr', 'subjectexpr'):
             if isinstance(self, CreateConstraint):
                 params = self._get_params(schema, context)
             else:
+                # TODO: this line is not hit in any test, is the assertion
```

Comment:
How can we cover this block with tests?

Reply:
We don't allow `ALTER`-ing constraint expressions currently, so the `else` branch should probably be an assertion error instead.

Reply:
I think in cases like this we shouldn't use assertions but an explicit `if not isinstance: raise`.

Reply:
Drawing the line where to put an `assert` expression and where to use an explicit exception is kinda hard though. In this case we have an `if` on one thing but check another thing in the `else`; the code isn't easy to read as result.

Reply:
@1st1, I followed your advice. I generally prefer to raise `if not instance` too, especially because assertions can be disabled when running Python with -O flag. But I have mixed feelings for the assertions used to pass mypy validation: assertions in this case look reasonable because it's something about the programmer rather than the end user.

In this specific case, I altered the code to raise `if not isinstance(self, CreateConstraint)`, since the "else block" was not tested and Elvis said that scenario is not supported.